### PR TITLE
Fix #52 streaming errors

### LINKED ISSUE #52: Errors in streaming.py
@yanchengnv notice some issues in nvflare/app_common/widgets/streaming.py:

- Line 47 to Line 52, the checking of the args and error messages are wrong.
- All these write_xxx() methods, should check the tag and data arg and make sure they are what we expect (str, dict, …)
- Line 257, in the call self.log_xxx(), we should set send_event=False; otherwise it may cause recursive events
- Since fed events are handled by a separate thread, there is a potential racing condition that a fed event could be fired after END_RUN event. In the Receiver code, we need to make sure to discard other events after END_RUN (and hence finalize) is done.

### PR DESCRIPTION
**Pull request by YuanTingHsieh**
Fixes #52 

- Add error checks
- Fix typos
- Add a boolean self._end in AnalyticsReceiver to indicate if END_RUN event has been fired already

### CODE REVIEW

**Review by yanchengnv — DISMISSED** [review]
Looks good to me.

**Review by yanchengnv — APPROVED** [review]
Looks good.

File: test/test_streaming.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import pytest
```
[the remark this answers is not in the document]
Comment (yhwen) [reply test/test_streaming.py:15]: added

**Review by yhwen — APPROVED** [review]
LGTM

**Pull request merged**